pyaedt/edb_core/edb_data/simulation_setup.py

```diff
@@ -0,0 +1,720 @@
+        self._add_frequency_sweep(sweep)
+        self._update_setup()
+        return sweep
+
+
+class EdbFrequencySweep(object):
+    """Manages EDB methods for frequency sweep.
+
+    Parameters
+    ----------
+    sim_setup : :class:`pyaedt.edb_core.edb_data.siwave_simulation_setup_data.SiwaveSYZSimulationSetup`
+    name : str, optional
+        Name of the frequency sweep.
+    edb_sweep_data : :class:`Ansys.Ansoft.Edb.Utility.SIWDCIRSimulationSettings`, optional
+        Edb object.
+    """
+
+    def __init__(self, sim_setup, frequency_sweep=None, name=None, edb_sweep_data=None):
+        self._sim_setup = sim_setup
+
+        if edb_sweep_data:
+            self._edb_sweep_data = edb_sweep_data
+            self._name = self._edb_sweep_data.Name
+        else:
+            if not name:
+                self._name = generate_unique_name("sweep")
+            else:
+                self._name = name
+            self._edb_sweep_data = self._pedb.simsetupdata.SweepData(self._name)
+            self.set_frequencies(frequency_sweep)
+
+    @property
+    def _edb_object(self):
+        return self._edb_sweep_data
+
+    @property
+    def _pedb(self):
+        """EDB."""
+        return self._sim_setup._pedb
+
+    @pyaedt_function_handler()
+    def _update_sweep(self):
+        """Update the sweep."""
+        self._sim_setup.delete_frequency_sweep(self)
+        self._sim_setup._add_frequency_sweep(self)
+        return
+
+    @property
+    def name(self):
+        """Name of the sweep."""
+        return self._edb_sweep_data.Name
+
+    @name.setter
+    def name(self, value):
+        self._edb_sweep_data.Name = value
+        self._update_sweep()
+
+    @property
+    def sweep_type(self):
+        """Sweep type."""
+        return
+
+    @property
+    def frequencies(self):
+        """List of frequency points."""
+        return list(self._edb_sweep_data.Frequencies)
+
+    @property
+    def adaptive_sampling(self):
+        """Whether adaptive sampling is used.
+
+        Returns
+        -------
+        bool
+            ``True`` if adaptive sampling is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.AdaptiveSampling
+
+    @property
+    def adv_dc_extrapolation(self):
+        """Flag indicating if advanced DC extrapolation is turned on.
+
+        Returns
+        -------
+        bool
+            ``True`` if advanced DC Extrapolation is used, ``False`` otherwise.
+
+        """
+        return self._edb_sweep_data.AdvDCExtrapolation
+
+    @property
+    def compute_dc_point(self):
+        """Flag indication if compute exact dc point is turned on."""
+        return self._edb_sweep_data.ComputeDCPoint
+
+    @compute_dc_point.setter
+    def compute_dc_point(self, value):
+        self._edb_sweep_data.ComputeDCPoint = value
+        self._update_sweep()
+
+    @property
+    def auto_s_mat_only_solve(self):
+        """Flag indication if Auto SMatrix only solve is turned on."""
```

Comment:
```suggestion
        """Flag indicating if Auto SMatrix only solve is turned on."""
```
The phrase ``if Auto SMatrix only solve`` needs to be rewritten in plain English.